std/xml.d

```diff
@@ -562,7 +562,7 @@ class Document : Element
         string tagString = xml.tag.tagString;
 
         this(xml.tag);
-        prolog = s[0 .. tagString.ptr - s.ptr];
+        prolog = s[0 .. &tagString[0] - &s[0]];
```

Comment:
>I'm surprised this makes a difference to the compiler

It doesn't. I still can't mark this ctor as safe because it calls parse. This just makes the function safer than it was before.